### PR TITLE
Add missing code example for Write::write_vectored

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: library/std/src/io/mod.rs [hunk 1]
```diff
@@ -1418,6 +1418,25 @@ pub trait Write {
...
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let mut data1 = [1; 8];
+    ///     let mut data2 = [15; 8];
+    ///     let mut io_slice1 = IoSlice::new(&mut data1);
+    ///     let mut io_slice2 = IoSlice::new(&mut data2);
+    ///
+    ///     let mut buffer = File::create("foo.txt")?;
+    ///
+    ///     buffer.write_vectored(&[io_slice1, io_slice2])?;
```
Comment (the8472) [inline library/std/src/io/mod.rs:1435]: The `write` example contains a comment highlighting that it's not guaranteed to write all data. Either the same should be done here too or the example should handle the return value.

Comment (GuillaumeGomez) [reply library/std/src/io/mod.rs:1435]: This is a good point. I'll update the example.

File: library/std/src/io/mod.rs [hunk 2]
```diff
@@ -1418,6 +1418,27 @@ pub trait Write {
...
+    /// ```no_run
+    /// use std::io::IoSlice;
+    /// use std::io::prelude::*;
+    /// use std::fs::File;
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let mut data1 = [1; 8];
+    ///     let mut data2 = [15; 8];
+    ///     let mut io_slice1 = IoSlice::new(&mut data1);
+    ///     let mut io_slice2 = IoSlice::new(&mut data2);
```
Comment (JohnTitor) [inline library/std/src/io/mod.rs:1432]: `bufs` doesn't have to be `mut`.

```suggestion
    ///     let io_slice1 = IoSlice::new(&mut data1);
    ///     let io_slice2 = IoSlice::new(&mut data2);
```

Comment (GuillaumeGomez) [reply library/std/src/io/mod.rs]: Thanks!

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
LGTM, let's go ahead.
r? @JohnTitor @bors r+ rollup

**Pull request merged**